packages/react-admin-core/src/AdminRouter.js

```diff
@@ -96,18 +96,7 @@ export class AdminRouter extends Component {
             typeof children === 'function' ? childrenFromState : children;
 
         if (!childrenToRender || childrenToRender.length === 0) {
-            return (
```

Comment:
this is a BC break, and your PR shouldn't break BC

Reply:
This is a BC break for custom apps only. Solution would be to have this renamed as `CoreAdminRouter` and have an `AdminRouter` setting the default `loading` inside `ra-ui-materialui`. It won't be used by default though.